app/src/main/java/com/example/android/codelabs/paging/ui/SearchRepositoriesActivity.kt

```diff
@@ -58,47 +58,41 @@ class SearchRepositoriesActivity : AppCompatActivity() {
         // add dividers between RecyclerView's row items
         val decoration = DividerItemDecoration(this, DividerItemDecoration.VERTICAL)
         binding.list.addItemDecoration(decoration)
-        setupScrollListener()
 
-        initAdapter()
+        binding.list.adapter = adapter.withLoadStateHeaderAndFooter(
+                header = ReposLoadStateAdapter { adapter.retry() },
+                footer = ReposLoadStateAdapter { adapter.retry() }
+        )
+        adapter.addLoadStateListener { loadType, loadState ->
+            Log.d("SearchRepositoriesActivity", "adapter load: type = $loadType state = $loadState")
+            if (loadType == LoadType.REFRESH) {
```

Comment:
I think it may be clearer if you put the entire body of this behind a "LoadType == Refresh" check.

That also makes the Toast only happen when it's a refresh error, instead of all

Reply:
I actually wanted to happen always.
In the initial code, whenever there is an error we show the toast.